### PR TITLE
fix(target): restrict target from IPRange to entity task if needed

### PR DESCRIPTION
**Pull request by stonebuzz**
revert https://github.com/glpi-project/glpi-inventory-plugin/pull/173

restrict target from IPRange to entity task if needed (add new option to use restrict)

see: https://github.com/glpi-project/glpi-inventory-plugin/issues/246

### CODE REVIEW

File: tests/Integration/Tasks/NetworkInventoryTest.php [hunk 1]
```diff
@@ -311,7 +289,7 @@ public function prepareDb()
         $ipranges_id = $pfIPRange->add($input);
         $this->assertNotFalse($ipranges_id);
 
-       // Allow all agents to do network inventory
+       // Allow all agents to do network discovery
```
Comment (trasher) [inline tests/Integration/Tasks/NetworkInventoryTest.php:292]: Seems it's doing inventory here, not discovery

Comment (stonebuzz) [reply tests/Integration/Tasks/NetworkInventoryTest.php]: done

**Review by trasher — APPROVED** [review]

**Review by orthagh — APPROVED** [review]

File: inc/taskjobview.class.php [hunk 2]
```diff
@@ -619,22 +619,57 @@ public function showForm($id, $options = [])
...
+        echo Html::getCheckbox($options);
+        echo Html::showToolTip(__('Only for IPRange, restrict target to task entity. Unchecked if assets are not in the same entity as the task'), ['display' => true]);
+        echo "</div>";
+
         echo "</div>"; // end of first inputs column wrapper
 
+
+
+
+
```
Comment (cedric-anne) [inline inc/taskjobview.class.php:672]:
```suggestion
```

File: inc/taskjobview.class.php [hunk 3]
```diff
@@ -619,22 +619,57 @@ public function showForm($id, $options = [])
         if (!is_null("method")) {
             $options['value'] = $this->fields["method"];
         }
+
+        $options["on_change"] = "task_methode_change(this.value)";
```
Comment (cedric-anne) [inline inc/taskjobview.class.php:623]:
```suggestion
        $options["on_change"] = "task_method_change(this.value)";
```

File: inc/taskjobview.class.php [hunk 4]
```diff
@@ -619,22 +619,57 @@ public function showForm($id, $options = [])
...
+
         $modules_methods_rand = Dropdown::showFromArray(
             "method",
             $modules_methods,
             $options
         );
         echo "</div>";
 
+        echo Html::scriptBlock("
+        function task_methode_change(val) {
```
Comment (cedric-anne) [inline inc/taskjobview.class.php:633]:
```suggestion
        function task_method_change(val) {
```

File: install/update.native.php [hunk 5]
```diff
@@ -791,6 +791,9 @@ function pluginGlpiinventoryUpdateNative($current_version, $migrationname = 'Mig
                 )
             );
         }
+
+        //new field
+        $migration->addField('glpi_plugin_glpiinventory_taskjobs', 'restrict_to_task_entity', "tinyint NOT NULL DEFAULT '1'", ['value'   => '1']);
```
Comment (cedric-anne) [inline install/update.native.php:796]:
```suggestion
        $migration->addField('glpi_plugin_glpiinventory_taskjobs', 'restrict_to_task_entity', 'bool', ['value'   => '1']);
```

**Review by cedric-anne — APPROVED** [review]

**Review by cedric-anne — APPROVED** [review]

**Pull request merged**